### PR TITLE
feat(demo): combined the modal examples

### PR DESCRIPTION
**Pull request by dmgauthier**
### Proposed Changes

Combined the modal examples and deleted deprecated stuff.

*TODO:* Will soon after add another example with a provider that prevent navigation when the modal has dirty changes

only affect the demo

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/components/drop/examples/DropExamples.tsx [hunk 1]
```diff
@@ -15,16 +14,10 @@ import {Drop} from '../Drop';
...
 
-const mapDispatchToProps = (dispatch: (action: IReduxAction<IModalActionPayload>) => void): IModalExamplesProps => ({
-    openModal: (id: string) => dispatch(openModal(id)),
-});
-
-@ReduxConnect(null, mapDispatchToProps)
 export class DropExamples extends React.PureComponent<any> {
     openModal(id: string) {
-        this.props.openModal(id);
+        ReactVaporStore.dispatch(openModal(id));
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/drop/examples/DropExamples.tsx]: I feel like the `connect` way is closer to reality

File: packages/react-vapor/src/components/modal/examples/modalWindowExamples.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+const ModalCompositeExample: React.FunctionComponent = () => {
+    const [isOpened, setOpened] = React.useState(false);
+
+    return (
+        <Section level={3} title="A modal window component with a local state">
+            <div>
+                <button className="btn" onClick={() => setOpened(true)}>
+                    Open Modal
+                </button>
+                <ModalComposite
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/modal/examples/modalWindowExamples.tsx]: I'm wondering if it's really useful to have a non-connected example

File: packages/react-vapor/src/components/drop/examples/DropExamples.tsx (hunk 1 above)
Comment (pochretien) [reply packages/react-vapor/src/components/drop/examples/DropExamples.tsx]: you can modify this file to use the new syntax with DropExamplesDisconnected and use connect below after :)

File: packages/react-vapor/src/components/modal/examples/modalWindowExamples.tsx (hunk 2 above)
Comment (dmgauthier) [reply packages/react-vapor/src/components/modal/examples/modalWindowExamples.tsx]: If you think there is really no use cases where we will want to use an unconnected modal, I'll remove it. :+1:

File: packages/react-vapor/src/components/drop/examples/DropExamples.tsx (hunk 1 above)
Comment (dmgauthier) [reply packages/react-vapor/src/components/drop/examples/DropExamples.tsx]: :+1:  :heart: realism

**Pull request merged**